### DISCUSSION TITLE
Query compilation cache when using engines created via Engine.execution_options()

### DISCUSSION THREAD

**Discussion by tonycosentini** [sqlalchemy/sqlalchemy#11454] [Usage Questions] [answered]
Hi,

I'm working on a multi-tenancy application that is going to leverage [`execution_options`](https://docs.sqlalchemy.org/en/20/core/connections.html#sqlalchemy.engine.Engine.execution_options) to create engines that use a schema translate map while still sharing a connection pool.

Is the query compilation also shared with the parent engine? I'm asking to gauge how much memory impact there might be with this.

Thanks!

**Comment by zzzeek** [comment, marked as answer]
if you make copies of engines with execution_options, that cache is shared; it's assigned in the constructor of `OptionEngine` to be the same dictionary as the parent.

**Reply by tonycosentini** [reply]
Amazing, thanks @zzzeek!
